manager/controllers/advisory_detail.go

```diff
@@ -53,28 +57,42 @@ func AdvisoryDetailHandler(c *gin.Context) {
 		return
 	}
 
+	resp, err := getAdvisory(advisoryName)
+	if err != nil {
+		if errors.Is(err, gorm.ErrRecordNotFound) {
+			LogAndRespNotFound(c, err, "advisory not found")
+		} else {
+			LogAndRespError(c, err, "advisory detail error")
+		}
+		return
+	}
+
+	c.JSON(http.StatusOK, resp)
+}
+
+func getAdvisoryFromDB(advisoryName string) (*AdvisoryDetailResponse, error) {
 	var advisory models.AdvisoryMetadata
 	err := database.Db.Where("name = ?", advisoryName).First(&advisory).Error
+	if err != nil {
+		return nil, err
+	}
+
 	if errors.Is(err, gorm.ErrRecordNotFound) {
-		LogAndRespNotFound(c, err, "advisory not found")
-		return
+		return nil, nil
 	}
 
 	if err != nil {
-		LogAndRespError(c, err, "database error")
```

Comment:
Same here, if `err != nil` you already returned from the function on L77.

Reply:
yes, thanks